src/main/java/edu/kit/compiler/semantic/ErrorQueue.java

```diff
@@ -0,0 +1,22 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.logger.Logger;
+import lombok.Getter;
+import lombok.NonNull;
+
+public class ErrorQueue {
+    @Getter
+    private boolean hasError;
+    @NonNull
+    private Logger logger;
+
+    public ErrorQueue(Logger logger) {
```

Comment:
Why isn't this queue a queue? Right now its just an object that logs a source location?

Reply:
Well, I can rename it to `ErrorHandler` or something